lib/read.d.ts

```diff
@@ -0,0 +1,20 @@
+/// <reference types="node" />
+import type { ReadLineOptions } from 'node:readline'
+
+declare function Read<T = string>(
```

Comment:
It might be safer to just always use `string` instead of inferring an alternate type from the `default` option, but I figured this was the least obtrusive way to do it.  So if you do `const x = await read({ def: 1 })` then `x` will be `string | number`.  Since the default option isn't forced to be a string in the code, this seems safest, and shouldn't complain if anyone is using it that way today.